reconcile/utils/jjb_client.py

```diff
@@ -293,7 +293,10 @@ def get_job_webhooks_data(self):
                     project_url = project_url_raw.strip("/").replace(".git", "")
                     gitlab_triggers = job["triggers"][0]["gitlab"]
                     mr_trigger = gitlab_triggers["trigger-merge-request"]
-                    trigger = "mr" if mr_trigger else "push"
+                    push_trigger = gitlab_triggers["trigger-push"]
+                    trigger = "mr" if mr_trigger else "push" if push_trigger else None
```

Comment:
nit: this logic is a little nested, can we flatten it like this:

```python
mr_trigger = "mr" if gitlab_triggers["trigger-merge-request"] else None
push_trigger = "push" if gitlab_triggers["trigger-push"] else None
trigger = mr_trigger or push_trigger
```

Reply:
a nit indeed

Reply:
i'll leave things the way they are for now if you don't mind. i personally find it more readable.

Reply:
@maorfr it wasn't very clear to me, I was wondering which one has higher priority, it turned out `trigger-merge-request` is higher than `trigger-push`, the nested one line `if else` hide it from me because I wasn't sure about the evaluation order.

For example, `A if B else C if D else E`, will `D` be evaluated before or after `B`, in this case `B` will run first, but it is not always first expression first run in other python expressions like comprehensions. I don't like nested expressions in general, quote zen of python `Flat is better than nested.`

To me, the best way to describe this logic is to extract a function like:

```python
def get_trigger(gitlab_triggers):
  if gitlab_triggers["trigger-merge-request"]:
    return "mr"
  if gitlab_triggers["trigger-push"]:
    return "push"
  return None
```

Reply:
i challenge you to find the real problem with the code :wink:

Reply:
also notice this integration changes so rarely, so there is a very low ROI here

Reply:
There are 2 ways of coding, do it complex so there is no obvious problem, or do it simple so there is obviously no problem. I prefer the later.

Reply:
you see, a cleaner approach will reveal this easily as `trigger = mr_trigger or push_trigger` will indicate there can be at most one of them, that's the assumption I made from reading original code

Reply:
If we want to create different triggers in different conditions, I suggest we extract a function

```python
def get_trigger(gitlab_triggers):
    triggers = []
    if gitlab_triggers["trigger-merge-request"]:
      triggers.append("mr")
    if gitlab_triggers["trigger-push"]:
      triggers.append("push")
    return triggers
```

then we can create iterate that to create triggers, it can handle no trigger case as well.

Reply:
I checked gitlab [add-project-hook api](https://docs.gitlab.com/ee/api/projects.html#add-project-hook), one webhook can support multiple triggers, so one webhook is enough here.

Instead of iterate on `triggers` to create multiple webhooks, we can check it like `mr in triggers` in https://github.com/app-sre/qontract-reconcile/blob/47761a04b50a07df8464a36dc34710102c23ba18/reconcile/utils/gitlab_api.py#L537-L551.

Btw it seems we only have `create_project_hook` but no `update` or `delete`, also create only in [jenkins_webhooks](https://github.com/app-sre/qontract-reconcile/blob/47761a04b50a07df8464a36dc34710102c23ba18/reconcile/jenkins_webhooks.py). So existing webhooks won't be synced.

Reply:
this is by design, to allow users to define their own webhooks. there is another integration to cleanup, but it's mostly for migrations.